console/client/src/features/modules/ModulesPage.tsx

```diff
@@ -1,11 +1,52 @@
+import React from 'react'
 import { Square3Stack3DIcon } from '@heroicons/react/24/outline'
 import { PageHeader } from '../../components/PageHeader'
+import { modulesContext } from '../../providers/modules-provider'
+import { Disclosure } from '@headlessui/react'
+import { createLayoutDataStructure } from './create-layout-data-structure'
 
 export const ModulesPage = () => {
+  const modules = React.useContext(modulesContext)
+  const data = createLayoutDataStructure(modules)
```

Comment:
<img width="278" alt="Screenshot 2023-09-16 at 7 51 14 AM" src="https://github.com/TBD54566975/ftl/assets/51647/c14fddd1-a116-4277-83ab-4910161e4c44">

I like the idea of this layout. I'm guessing this is intended to display the dependency graph similar to our `Graph` page.  

<img width="1003" alt="Screenshot 2023-09-16 at 7 58 37 AM" src="https://github.com/TBD54566975/ftl/assets/51647/932b502e-d9f1-44f9-9329-183421f0a9a6">

This graph helps me view the dependencies (even though our layout algorithm isn't great :)) I see with this graph that `productCatalog` depends on `recommendation` too. Maybe `recommendation` should be under `productCatalog`.

Given that multiple modules can depend on the same module (maybe that's not the case with our boutique demo but it's possible) we might end up with duplicate module nodes here. An example that comes to mind is, what if `cart` also used `recommendation`? Then would we have a `recommendation` node below `cart` and one below `productCatalog`?

Reply:
Updated description on this PR please see that should address questions